src/functions.rs

```diff
@@ -2740,6 +2750,22 @@ fn validate_argument<'a>(
             if arg_typeinfo.is_child_or_actual_type(target_argument.param_type, types) {
                 Ok(TypeInstance::new(&arg, arg_typeinfo, file, context))
             } else {
+                if let Some(file) = file {
+                    if file.name().is_empty() {
+                        return Err(ErrorItem::make_compile_or_internal_error(
+                            &format!(
+                                "Expected type inheriting {} for associated call",
+                                target_argument.param_type.name
+                            ),
+                            arg_typeinfo.get_file().as_ref(),
+                            arg_typeinfo.name.get_range(),
+                            &format!(
+                                "An associated call was made for this domain.  This type should inherit {}",
+                                target_argument.param_type.name
+                            ),
+                        ));
+                    }
+                }
```

Comment:
This is not as good as it should be.  Right now, we are just pointing to the domain definition and saying it needs to inherit X.  But I am struggling with the best way to get information about the associated call/class here.  Any guidance would be appreciated.

Reply:
I think most of the issue here is related to the discussion on the other comment.  A lot of what you need will be in the FunctionInfo.  And we should know the domain with the call from the context, I imagine, right?  Between the context and FunctionInfo, I'd think we should end up with enough data to write a helpful error message, once we have the FunctionInfo available at the message spot.